Framework/PythonInterface/plugins/functions/SpinDiffusion.py

```diff
@@ -0,0 +1,93 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+# pylint: disable=no-init,invalid-name
+import numpy as np
+from functools import cache
+from numpy import float64
+from numpy.typing import NDArray
+from mantid.api import IFunction1D, FunctionFactory
+from mantid.kernel import IntBoundedValidator
+from mantid.kernel import PhysicalConstants
+from scipy.integrate import quad
+from scipy.special import i0e
+from typing import Tuple
+
+MUON_MAGNETOGYRIC_RATIO = 2 * np.pi * PhysicalConstants.MuonGyromagneticRatio  # MHz / G
+
+
+@cache
```

Comment:
This is a nice touch, to cache the result so not duplicating at least one calculation (depending on dimensionality)! I would probably have stored the values in a member variable but maybe this is quicker (it's certainly cleaner).